extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/token/propagation/TokenPropagationConstants.java

```diff
@@ -0,0 +1,21 @@
+package io.quarkus.oidc.token.propagation;
+
+public final class TokenPropagationConstants {
+
+    TokenPropagationConstants() {
+    }
+
+    /**
+     * System property key that is resolved to true if OIDC auth mechanism should put
+     * `TokenCredential` into Vert.x duplicated context.
+     */
+    public static final String OIDC_PROPAGATE_TOKEN_CREDENTIAL = "io.quarkus.oidc.runtime." +
+            "AbstractOidcAuthenticationMechanism.PROPAGATE_TOKEN_CREDENTIAL_WITH_DUPLICATED_CTX";
+    /**
+     * System property key that is resolved to true if JWT auth mechanism should put
+     * `TokenCredential` into Vert.x duplicated context.
+     */
+    public static final String JWT_PROPAGATE_TOKEN_CREDENTIAL = "io.quarkus.smallrye.jwt.runtime." +
```

Comment:
It ts probably a copy/paste problem, this property is not used

Reply:
it is used when it is produced in 2 build steps (classic / reactive token propagation) and then it is used in both access token filters. I have reasons for that -> we can't rely on that buildtime property outside of build time because built time properties are not kept for runtime (imho it changed with Quarkus 3) and I don't want to introduce other new runtime property, nor I want to set it after CDI initialization on beans because none of token propagation extensions have recorder. It feels right to keep it simple.

Reply:
btw idea to use system property this way came to me when I read `VertxContextSafetyToggle` class. I want to avoid introduction of multiple config properties that for now we should have control (there it is more for build step), but avoid new SPI

Reply:
@michalvavrik Sorry, I meant that `OIDC_PROPAGATE_TOKEN_CREDENTIAL` is indeed what you use in reactive/classic token propagation filters, `JWT_PROPAGATE_TOKEN_CREDENTIAL` is not used anywhere since `quarkus-smallrye-jwt` does not depend on `quarkus-oidc-client`

Reply:
@sberyozkin please see:

- https://github.com/quarkusio/quarkus/blob/202addd77bc472eb37a474c31063dc7d783c7b5a/extensions/oidc-token-propagation-reactive/runtime/src/main/java/io/quarkus/oidc/token/propagation/reactive/AccessTokenRequestReactiveFilter.java#L57
- https://github.com/quarkusio/quarkus/blob/202addd77bc472eb37a474c31063dc7d783c7b5a/extensions/oidc-token-propagation/deployment/src/main/java/io/quarkus/oidc/token/propagation/deployment/OidcTokenPropagationBuildStep.java#L54
- https://github.com/quarkusio/quarkus/blob/202addd77bc472eb37a474c31063dc7d783c7b5a/extensions/oidc-token-propagation/runtime/src/main/java/io/quarkus/oidc/token/propagation/AccessTokenRequestFilter.java#L50
- https://github.com/quarkusio/quarkus/blob/202addd77bc472eb37a474c31063dc7d783c7b5a/extensions/oidc-token-propagation-reactive/deployment/src/main/java/io/quarkus/oidc/token/propagation/reactive/OidcTokenPropagationReactiveBuildStep.java#L68

Reply:
@michalvavrik Oh yes, sorry, of course the propagation code needs to be aware of the property set by either smallryw-jwt or OIDC - note you can't combine both smallrye-jwt and OIDC bearer mechanisms - and if it were only about them we could've used a single property, but, OIDC code flow can be combined with smallrye-jwt bearer, so indeed 2 properties have to be used, thanks